app/controllers/communicarts_controller.rb

```diff
@@ -52,6 +55,9 @@ def approval_response
 private
 
   def validate_access
+    request_from_email = params['email_delivery'].presence || true
+    return unless request_from_email == true
```

Comment:
`return if params['email_delivery'] == false` is more readable to my untrained eye.

Bigger question: why is whether or not a user came from email the deciding factor? I'd expect the token to be ignored (though, perhaps still expired) if the user was already logged in, regardless of how the user reached this page.